core/src/main/java/com/linecorp/armeria/common/TlsSetters.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.File;
+import java.io.InputStream;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.util.function.Consumer;
+
+import javax.net.ssl.KeyManagerFactory;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+import io.netty.handler.ssl.SslContextBuilder;
+
+/**
+ * Sets properties for TLS or SSL.
+ */
+@UnstableApi
+public interface TlsSetters {
+
+    /**
+     * Configures SSL or TLS with the specified {@code keyCertChainFile}
+     * and cleartext {@code keyFile}.
+     */
+    default TlsSetters tls(File keyCertChainFile, File keyFile) {
```

Comment:
Maybe add some one-liner stating that this is for client cert authentication, ie. has nothing to do with authenticating the server cert and everything to do with authentication the client?

Reply:
Actually, this can be used for both server and client authentication. But it is worth overriding the Java in `ClientFactoryBuilder`.